aws-lambda/massdriver.yaml

```diff
@@ -7,27 +7,80 @@ type: application
 
 params:
   required:
-  - region
-  - port
-  - log_level
+    - aws_region
   properties:
-    location:
-      title: Region
-      # Massdriver defines many reusable types: https://github.com/massdriver-cloud/artifact-definitions/tree/main/definitions/types
+    aws_region:
+      title: AWS Region 
       $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/aws-region.json
-      $md.immutable: true
-    port:
-      title: Port
-      description: "The port number to run on. This is an example value, this could be anything!"
-      type: string
-    log_level:
-      type: string
-      title: Log Level
-      description: "The log level to log at. This is an example value, this could be anything!"
-      enum:
-        - error
-        - warning
-        - info
+    runtime:
+      type: object
+      title: Runtime Settings
+      required:
+        - image
+        - memory_size
+        - execution_timeout
+      properties:
+        image:
+          type: object
+          title: "Container Image"
+          required:
+            - uri
+            - tag
+          properties:
+            uri:
+              title: ECR URI
+              description: Container which will provide the runtime for each execution of the function.
+              type: string
+            tag:
+              title: ECR image tag
+              description: Specific version of the container to pull on each deployment of the lambda function.
+              type: string
+              default: latest
+        memory_size:
+          type: integer
+          title: Runtime Memory Limit (MB)
+          description: "Memory allotted for a single execution run. This will determine the price per millisecond."
+          default: 128
+          minimum: 128
+          maximum: 10240
+        execution_timeout:
+          type: integer
+          title: Execution Timeout (Seconds)
+          description: "Maximum execution time. After the allotted time, the lambda will halt execution."
+          default: 3
+          minimum: 3
+          maximum: 900
+    observability:
+      type: object
+      title: "Observability Settings"
+      required:
+        - retention_days
+      properties:
+        x-ray:
+          type: object
+          title: X-Ray
```

Comment:
May want to title it Enable X-Ray since MUI still doesn’t show checkbox desc

Reply:
I nested enabled under x-ray so we could use that object description field due to that checkbox limitation. Thoughts?

Reply:
This is a clever way around the issue. Only 2 things to consider:
* If we fix the boolean description issue, this will require a DB migration to move it out of an object, if we choose to
* In the UI, is the boolean checkbox separated at all from the Object definition enough to make it unclear with the boolean is?  There is no title so it uses the field name (`enabled`)

It seems to look alright: [link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaXJzdE5hbWUiOiJDaHVjayIsImZvbyI6e30sImxhc3ROYW1lIjoiTm9ycmlzIiwiYWdlIjo3NSwiYmlvIjoiUm91bmRob3VzZSBraWNraW5nIGFzc2VzIHNpbmNlIDE5NDAiLCJwYXNzd29yZCI6Im5vbmVlZCJ9LCJzY2hlbWEiOnsidGl0bGUiOiJBIHJlZ2lzdHJhdGlvbiBmb3JtIiwiZGVzY3JpcHRpb24iOiJBIHNpbXBsZSBmb3JtIGV4YW1wbGUuIiwidHlwZSI6Im9iamVjdCIsInJlcXVpcmVkIjpbImZpcnN0TmFtZSIsImxhc3ROYW1lIl0sInByb3BlcnRpZXMiOnsiZmlyc3ROYW1lIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6IkZpcnN0IG5hbWUiLCJkZWZhdWx0IjoiQ2h1Y2sifSwibGFzdE5hbWUiOnsidHlwZSI6InN0cmluZyIsInRpdGxlIjoiTGFzdCBuYW1lIn0sInRlbGVwaG9uZSI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJUZWxlcGhvbmUiLCJtaW5MZW5ndGgiOjEwfSwiZm9vIjp7InR5cGUiOiJvYmplY3QiLCJ0aXRsZSI6IlgtUmF5IiwiZGVzY3JpcHRpb24iOiJFbmFibGUgZGlzdHJpYnV0ZWQgdHJhY2luZyB3aXRoIEFtYXpvbiBYLVJheSIsInByb3BlcnRpZXMiOnsiZW5hYmxlZCI6eyJ0eXBlIjoiYm9vbGVhbiIsImVudW0iOlt0cnVlLGZhbHNlXX19fX19LCJ1aVNjaGVtYSI6eyJmaXJzdE5hbWUiOnsidWk6YXV0b2ZvY3VzIjp0cnVlLCJ1aTplbXB0eVZhbHVlIjoiIiwidWk6YXV0b2NvbXBsZXRlIjoiZmFtaWx5LW5hbWUifSwibGFzdE5hbWUiOnsidWk6ZW1wdHlWYWx1ZSI6IiIsInVpOmF1dG9jb21wbGV0ZSI6ImdpdmVuLW5hbWUifSwiYWdlIjp7InVpOndpZGdldCI6InVwZG93biIsInVpOnRpdGxlIjoiQWdlIG9mIHBlcnNvbiIsInVpOmRlc2NyaXB0aW9uIjoiKGVhcnRoaWFuIHllYXIpIn0sImJpbyI6eyJ1aTp3aWRnZXQiOiJ0ZXh0YXJlYSJ9LCJwYXNzd29yZCI6eyJ1aTp3aWRnZXQiOiJwYXNzd29yZCIsInVpOmhlbHAiOiJIaW50OiBNYWtlIGl0IHN0cm9uZyEifSwiZGF0ZSI6eyJ1aTp3aWRnZXQiOiJhbHQtZGF0ZXRpbWUifSwidGVsZXBob25lIjp7InVpOm9wdGlvbnMiOnsiaW5wdXRUeXBlIjoidGVsIn19fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZSI6ZmFsc2UsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZX19)